README.md

```diff
@@ -38,90 +38,133 @@
+
+``` python
+import tensorflow as tf
+import stellargraph as sg
+import pandas as pd
+from sklearn import model_selection
+
+#### Data preparation ####
+
+# Use Pandas to load and preprocess the data
+nodes = pd.read_csv("nodes.csv") # columns of node data
+edges = pd.read_csv("edges.csv") # "source" & "target" columns for each link between nodes
+
+node_classes = pd.read_csv("labels.csv") # a column of each node's class/label
+targets = pd.get_dummies(node_classes) # one-hot encoding
 
-To get started with StellarGraph you'll need data structured as a homogeneous or heterogeneous graph, including
-attributes for the entities represented as graph nodes.
-[NetworkX](https://networkx.github.io/) is used to represent the graph and [Pandas](https://pandas.pydata.org/)
-or [Numpy](https://www.numpy.org/) are used to store node attributes.
+# Use scikit-learn to compute training and test sets
+train_targets, test_targets = model_selection.train_test_split(targets, train_size=0.5)
 
-Detailed and narrated [examples](https://github.com/stellargraph/stellargraph/tree/master/demos/) of various machine learning workflows on network data, supported by StellarGraph, from data ingestion into graph structure to inference, are given in the `demos` directory of this repository.
+#### Graph machine learning model ####
 
-<!--
-StellarGraph supports different machine learning use-cases, including:
+# Use StellarGraph to create the graph machine learning model: graph, data generator, and GCN layers
+graph = sg.StellarGraph(nodes, edges)
 
-* Representation learning for nodes
-  - See the demos in folder `demos/embeddings` for examples of unsupervised node representation learning using the
-  random walk-based methods Node2Vec [1], and Metapath2Vec [2].
+generator = sg.mapper.FullBatchNodeGenerator(graph, method="gcn")
+train_gen = generator.flow(train_targets.index, train_targets)
 
-* Node classification and regression
-  - See the demo in folder `demos/node-classification-graphsage` for an example of how to predict attributes of nodes
-  using the GraphSAGE [3] algorithm given node features and training labels.
-  - See the demo in folder `demos/node-classification-node2vec` for an example of how to predict attributes of nodes
-  using the Node2Vec [1] algorithm for nodes without features, unsupervised node representation learning, and
-  supervised classifier training for the downstream task.
-  - See the demo in folder `demos/node-classification-hinsage` for examples of how to predict attributes of nodes
-  using the HinSAGE algorithm for given node features and training labels.
+# two layers of GCN, each with hidden dimension 16
+gcn = sg.layer.GCN(
+    layer_sizes=[16, 16], activations=["relu", "relu"], generator=generator, dropout=0.5
+)
```

Comment:
Thoughts on removing some of the optional parameters here? I think `relu` should be default already
```suggestion
gcn = sg.layer.GCN(layer_sizes=[16, 16], generator=generator)
```

Reply:
Or even
```suggestion
x_inp, x_out = sg.layer.GCN(layer_sizes=[16, 16], generator=generator).build()
```

Reply:
I like it, although I think I'd prefer to have `build` be separate.